File: solutions/java/high-scores/1/src/main/java/HighScores.java
import java.util.*;

class HighScores {
    private final List<Integer> highScores;
    
    public HighScores(List<Integer> highScores) {
        this.highScores = new ArrayList<>(highScores);
    }

    List<Integer> scores() {
        return new ArrayList<>(highScores); 
    }

    Integer latest() {
        return highScores.get(highScores.size() - 1);
    }

    Integer personalBest() {
        return Collections.max(highScores);
    }

    List<Integer> personalTopThree() {
        return highScores.stream()
                         .sorted(Comparator.reverseOrder())
                         .limit(3)
                         .toList();
    }

}

File: solutions/javascript/high-scores/1/high-scores.js
//
// This is only a SKELETON file for the 'High Scores' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class HighScores {
  #list;
  
  constructor(list) {
    this.#list = list 
  }

  get scores() {
    return this.#list;
  }

  get latest() {
    return this.#list.at(-1);
  }

  get personalBest() {
    return Math.max(...this.#list);
  }

  get personalTopThree() {
     
    return [...this.#list]
    .sort((a,b)=> b-a).slice(0,3);
  }
}
